### PR TITLE
2021-06-01 version for AzureADOnlyauthSupport

### PR DESCRIPTION
**Pull request by sravan251**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [x] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [x] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [x] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by idear1203** [comment]
1. Suggest to use [OpenAPIHub](https://aka.ms/openapihub) to create Swagger spec for new API versions.
2. Is this a GA feature? If not, why the API version is a stable API version (2021-06-01 without -preview)? If so, please make sure there is no breaking change going forward.
3. Is the API deployment at server side already done?

**Comment by sravan251** [comment]
Suggest to use OpenAPIHub to create Swagger spec for new API versions.
Is this a GA feature? If not, why the API version is a stable API version (2021-06-01 without -preview)? If so, please make sure there is no breaking change going forward.
Is the API deployment at server side already done?
Please fix CI validation issues

> 1. Suggest to use [OpenAPIHub](https://aka.ms/openapihub) to create Swagger spec for new API versions.
> 2. Is this a GA feature? If not, why the API version is a stable API version (2021-06-01 without -preview)? If so, please make sure there is no breaking change going forward.
> 3. Is the API deployment at server side already done?
> 4. Please fix CI validation issues

Suggest to use OpenAPIHub to create Swagger spec for new API versions. Sure , From next time i will use openAPI Hub.
Is this a GA feature? If not, why the API version is a stable API version (2021-06-01 without -preview)? If so, please make sure there is no breaking change going forward.-We are adding a option field to support AzureADonlyauth similar to Publicnetworksaccess and CMK. So this is not required to be a preivew version. 
Is the API deployment at server side already done?Not yet. We will merge only after the Code is deployed.
Please fix CI validation issues. Only goSDK is f ailing with some issues. Need some help on this.

**Comment by idear1203** [comment]
@ruowan  this is to add azureADOnlyAuthentication.json in a new API version. Not sure why it fails breaking change check. Any thought? 

Besides, this PR is ready for a review. Could you please help with a review?

**Comment by ruowan** [comment]
@ArcturusZhang Could you help to take a look at go SDK breaking change?

**Comment by mentat9** [comment]
Please complete and correct the submission checklist. This is very important to get a correct and timely review.

Current issues with the checklist that need to be addressed: 
1. If this is a new API version and you have a previous API version, you need to follow this requirement (and check the box): **Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.**
2. Contribution checklist section is not filled in.
3. Documentation review requirement is not met: **Please ensure you've reviewed following guidelines including ARM resource provider contract and REST guidelines. Estimated time (4 hours). This is required before you can request review from ARM API Review board.**

**Comment by idear1203** [comment]
@sravan251 can we support the feature in the latest **preview** API version (or create a new preview API version based on the latest preview API version in order to pass the breaking change check) too? Swagger spec path: https://github.com/Azure/azure-rest-api-specs/tree/master/specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-06-01-preview . 

Reason: We want to make sure that the latest preview API version includes all Synapse APIs. Otherwise, in future, when we build SDK based on the latest preview API version, features only in stable API versions will be lost.  Same issue will also occur when we 
 create a new stable API version based on some existing latest preview API version.

**Comment by sravan251** [comment]
> @sravan251 can we support the feature in the latest **preview** API version (or create a new preview API version based on the latest preview API version in order to pass the breaking change check) too? Swagger spec path: https://github.com/Azure/azure-rest-api-specs/tree/master/specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-06-01-preview .
> 
> Reason: We want to make sure that the latest preview API version includes all Synapse APIs. Otherwise, in future, when we build SDK based on the latest preview API version, features only in stable API versions will be lost. Same issue will also occur when we
> create a new stable API version based on some existing latest preview API ver
 AzureAD auth and Storage PE support are optional features and not required to be part of 2021-06-01-preview. 2021-06-01-preview has additional resources which are not supported in stable version.

File: specification/synapse/resource-manager/Microsoft.Synapse/stable/2021-06-01/workspace.json
Comment (pilor) [inline specification/synapse/resource-manager/Microsoft.Synapse/stable/2021-06-01/workspace.json:1303]: how does this property on the workspace interact with the azureAdOnlyAuthentication child resource?  If I change this does the child resource change?  What about vice-versa?  If a child resource change changes this how do you notify downstream services like ARG that this resource has changed?

Comment (kashyap-kunal) [reply specification/synapse/resource-manager/Microsoft.Synapse/stable/2021-06-01/workspace.json:1303]: azureADOnlyAuthentication is more of a workspace setttings, then a child resource. User can modify its value either through PUT workspace call or PUT azureADOnlyAuthentication call.

In any of the case, Synapse updates the setting in its metastore and make a necessary RP call to partner service to reflect the change.

**Review by ruowan — APPROVED** [review]

**Pull request merged**
